app/controllers/order_items_controller.rb

```diff
@@ -0,0 +1,46 @@
+class OrderItemsController < ApplicationController
+  respond_to :html, :json
+  before_action :set_order, only: [:create, :destroy]
+
+  def create
+    @order_item = @order.add_book(order_item_params)
+    flash_msg(@order_item.save)
+    @order.valid?
+    redirect_back(fallback_location: books_path)
+  end
+
+  def destroy
+    @order_item = @order.order_items.find(params[:id])
+    @order_item.destroy
+    @order_items = @order.order_items
+    redirect_to cart_path
+  end
+
+  def update
+    order_item = OrderItem.find(params[:id])
+    if order_item_params[:quantity].present?
```

Comment:
All this quantity update code until `respond_to` can be extracted to model method or a service.